### PR TITLE
[ecosystem] Create IT1 registration form.

### PR DESCRIPTION
**Pull request by zacharydenton**
## Motivation

We need a website that allows participants to sign up for IT1 by logging in with GitHub and providing their mainnet address. This PR is the initial implementation of the frontend for that website. There are important features missing at this point, notably 1) integration with the yet-to-be-completed server-side 2) observability 3) non-placeholder CSS/design/copy.

onboarding_425

## Test Plan

I wrote automated tests.

### CODE REVIEW

File: ecosystem/platform/client/src/ui/SocialLoginButton.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+function oauthUrl(service: Service, csrfToken: string): string {
+  const clientId = CLIENT_IDS[service];
+  switch (service) {
+    case "github":
+      return `https://github.com/login/oauth/authorize?scope=user:email&client_id=${clientId}&state=${csrfToken}`;
+    default:
+      return assertNever(service);
+  }
+}
+
```
Comment (CapCap) [inline ecosystem/platform/client/src/ui/SocialLoginButton.tsx:25]: this is the code I've been using to drive the oauth flow:
```
cookieStore.get("CSRF-TOKEN").then((token)=>{
    var myForm = document.createElement("FORM");
    myForm.setAttribute("id","TestForm");
    document.body.appendChild(myForm);
    myForm.method = "POST";
    myForm.action = "/users/auth/github?authenticity_token=" + token.value;
    myForm.submit();
 });
 ```

Comment (zacharydenton) [reply ecosystem/platform/client/src/ui/SocialLoginButton.tsx:25]: Yeah, much of this will need to be updated once the server-side code is in.

**Review by davidiw — APPROVED** [review]
looks overall good for me,

two comments:
* Please add license headers to everything
* why no css?

**Comment by zacharydenton** [comment]
> Please add license headers to everything

Done.

> why no css?

I do have some basic CSS here (using [Tailwind](https://tailwindcss.com/)) but I don't know exactly how the final UI is going to look yet and until then I'm not putting much effort into the CSS.

**Pull request merged**